core/src/main/java/org/mskcc/cbio/portal/scripts/ImportCnaDiscreteLongData.java

```diff
@@ -0,0 +1,331 @@
+import java.util.stream.*;
+
+import static com.google.common.collect.Lists.*;
+import static java.lang.String.*;
+
+public class ImportCnaDiscreteLongData {
+
+    private final File cnaFile;
+    private final int geneticProfileId;
+    private final GeneticAlterationImporter geneticAlterationGeneImporter;
+    private String genePanel;
+    private final DaoGeneOptimized daoGene;
+    private CnaUtil cnaUtil;
+    private Set<CnaEvent.Event> existingCnaEvents = new HashSet<>();
+    private int samplesSkipped = 0;
+
+    private final ArrayList<SampleIdGeneticProfileId> sampleIdGeneticProfileIds = new ArrayList<>();
+
+    public ImportCnaDiscreteLongData(
+        File cnaFile,
+        int geneticProfileId,
+        String genePanel,
+        DaoGeneOptimized daoGene,
+        DaoGeneticAlteration daoGeneticAlteration
+    ) {
+        this.cnaFile = cnaFile;
+        this.geneticProfileId = geneticProfileId;
+        this.genePanel = genePanel;
+        this.daoGene = daoGene;
+        this.geneticAlterationGeneImporter = new GeneticAlterationImporter(geneticProfileId, daoGeneticAlteration);
+    }
+
+    public void importData() throws Exception {
+        FileReader reader = new FileReader(this.cnaFile);
+        BufferedReader buf = new BufferedReader(reader);
+
+        // Pass first line with headers to util:
+        String line = buf.readLine();
+        int lineIndex = 1;
+        this.cnaUtil = new CnaUtil(line);
+
+        GeneticProfile geneticProfile = DaoGeneticProfile.getGeneticProfileById(geneticProfileId);
+
+        boolean isDiscretizedCnaProfile = geneticProfile != null
+            && geneticProfile.getGeneticAlterationType() == GeneticAlterationType.COPY_NUMBER_ALTERATION
+            && geneticProfile.showProfileInAnalysisTab();
+
+        if (isDiscretizedCnaProfile) {
+            existingCnaEvents.addAll(DaoCnaEvent.getAllCnaEvents());
+            MySQLbulkLoader.bulkLoadOn();
+        }
+
+        CnaImportData toImport = new CnaImportData();
+
+        while ((line = buf.readLine()) != null) {
+            lineIndex++;
+            ProgressMonitor.incrementCurValue();
+            ConsoleUtil.showProgress();
+            this.extractDataToImport(geneticProfile, line, lineIndex, toImport);
+        }
+
+        DaoGeneticProfileSamples.addGeneticProfileSamples(
+            geneticProfileId,
+            newArrayList(toImport.eventsTable.columnKeySet())
+        );
+
+        for (Long entrezId : toImport.eventsTable.rowKeySet()) {
+            boolean added = storeGeneticAlterations(toImport, entrezId);
+            if (added) {
+                storeCnaEvents(toImport, entrezId);
+            } else {
+                ProgressMonitor.logWarning("Values not added to gene with entrezId: " + entrezId + ". Skip creation of cna events.");
+            }
+        }
+
+        ProgressMonitor.setCurrentMessage(" --> total number of samples skipped (normal samples): " + getSamplesSkipped()
+        );
```

Comment:
);

Reply:
I don't really want to wait for tests again, so just fix this in the next PR. I don't really think the `);` deserves its own line.